Added New Material System Documentation (#1730) (#1774)
o3de/o3de.org: content/docs/atom-guide/look-dev/materials/material-type-file-spec.md

  * **displayName**: The given name of this group that will appear in the Material Editor. 
  * **description**: The given description of this group that will appear as tooltips in the Material Editor.
  * **shaderInputsPrefix**: Any shader input names appearing in this group, such as shader constants or shader images, will automatically prepend this value. This is useful when using `$import` to include a shared group definitions in another group.
  * **shaderOptionsPrefix**: Any shader option names that appear in this group will automatically prepend this value. This is useful when using `$import` to include a shared group definitions in another group.
  * **properties**: Defines the list of properties that will appear in this group.
    * **name**: An identifier for this property. The value must be formatted in C-style and be unique among sibling properties. 
    * **displayName**: The given name of this property that will appear in the Material Editor. 
    * **description**: The given description of this property that will appear as tooltips in the Material Editor.

>     * **visibility**: The initial visibility for this property. Default is Enabled. Possible values are:

Reviewer: "Enabled" is the literal enum name of the default value. The default value is "Enabled".

>     * **visibility**: The initial visibility for this property. Default is Enabled. Possible values are:

Author: Thanks santorac, will fix this, and use code style so it's clear.

                            "type": "ShaderOption",
                            "name": "o_applySpecularAA"
                        }
                    },
                    {
                        "name": "enableMultiScatterCompensation",
                        "displayName": "Multiscattering Compensation",
                        "description": "Whether to enable multiple scattering compensation.",
                        "type": "Bool",
                        "connection": {
                            "type": "ShaderOption",
                            "name": "o_enableMultiScatterCompensation"
                        }
                    }
                ]
            }
        ]
    },
...
```

## **version** (deprecated)
This version number in the `propertyLayout` section is no longer used and has been replaced by the [`version`](#version) number above.

## **groups** (deprecated)
This is replaced by `propertyGroups` (in [propertyLayout](#propertylayout)), which can support any number of group levels with properties and groups defined in the same place. The engine can still load the old format but any new material type files should use the new format.

{{< note >}}

> An older version of the material type file format organized propertyLayout with separate `groups` and `properties` sections. This imposed several limitations like restricting property grouping to only one level and making it difficult to factor out reusable property groups.

Author: Removed the comma because "restricting property grouping to only one level" describes the whole limitation. Otherwise, with the comma, it seems to indicate that "to only one level" applies to "This imposed several limitations". 

```suggestion
An older version of the material type file format organized `propertyLayout` with separate `groups` and `properties` sections. This imposed several limitations, such as restricting property grouping to only one level. This made it difficult to factor out reusable property groups.
```
